api/nodes/utils.py

```diff
@@ -0,0 +1,23 @@
+import itertools
+
+from modularodm import Q
+
+from website.util import permissions as osf_permissions
+from website.models import Node
+
+def get_visible_nodes_for_user(user, include_public=True):
```

Comment:
I'm a bit inclined to leave this here until we identify specific needs for it elsewhere (although agreed we should unify our implementations of this kind of method).